### ISSUE TITLE
git-ext: Better handling of "not found" errors

### ISSUE THREAD

**Issue by cloudhead** [radicle-dev/radicle-git#29]
Currently I have code like this:
```rust
        match repo.reference(remote, sigref) {
            Err(git_ext::Error::Git(e)) if git_ext::is_not_found_err(&e) => None,
            Err(git_ext::Error::NotFound(_)) => None,
            Err(e) => return Err(e.into()),
...
        };
```
It would be nice if `git_ext::Error` simply had a `is_not_found(&self) -> bool` method, so that it could be converted to:
```rust
        match repo.reference(remote, sigref) {
            Err(e) if e.is_not_found() => None,
            Err(e) => return Err(e.into()),
...
        };
```
This is the pattern I've been using with my other error types.

**Comment by FintanH** [comment]
I'm a bit confused. There's no `git_ext::Error`?

There's one for [`blob.rs`](https://github.com/radicle-dev/radicle-git/blob/main/radicle-git-ext/src/blob.rs#L13-L21) and one for [`name.rs`](https://github.com/radicle-dev/radicle-git/blob/main/radicle-git-ext/src/reference/name.rs#L20-L28). Neither of those pertain to a function called `reference`.

However, we often do this:
```rust
use std_ext::ResultExt as _;
use git_ext as ext;

repo.reference(remote, sigref).or_matches(ext::is_not_found, || None)
```

**Comment by cloudhead** [comment]
Yes, it's re-exported from the top-level crate, so I know it as `git_ext::Error` :D

**Comment by cloudhead** [comment]
In that case I think we could create our own Git error type inside heartwood, and use that. No need to change anything with git_ext, just not export the errors from the top level, or alias them as eg. BlobError etc.

**Comment by cloudhead** [comment]
Actually it would still be useful to have that `is_not_found` method, for when we use the git_ext blob functions.
